publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java

```diff
@@ -65,27 +68,29 @@ public CreatePublicationHandler(ResourceService publicationService,
         this.apiHost = environment.readEnv(API_HOST);
         this.identityServiceClient = identityServiceClient;
     }
-    
+
     @Override
     protected PublicationResponse processInput(CreatePublicationRequest input, RequestInfo requestInfo,
                                                Context context) throws ApiGatewayException {
         logger.info(attempt(() -> JsonUtils.dtoObjectMapper.writeValueAsString(requestInfo)).orElseThrow());
-        
-        var userInstance = createUserInstanceFromLoginInformation(requestInfo);
+        if (isThesisAndHasNotPublishThesis(requestInfo, input)) {
```

Comment:
*AndHasNotPublishThesis is not the best method name. You can do something like 

hasRightsToPublishThesis(input, requestInfo);

or see comment below